modules/runner/src/integrationTest/java/org/apache/ignite/internal/table/ItInternalTableTest.java

```diff
@@ -384,78 +359,190 @@ public void testRoScanAllExplicitPopulatingData() throws InterruptedException {
-                retrievedItems.add(item);
-            }
+            if (i % 2 == 1) {
+                assertNull(resRow);
+            } else {
+                assertNotNull(resRow);
 
-            @Override
-            public void onError(Throwable throwable) {
-                fail("onError call is not expected.");
-            }
+                var rowTuple = TableRow.tuple(Row.wrapBinaryRow(schemaDescriptor, resRow));
 
-            @Override
-            public void onComplete() {
-                subscriberAllDataAwaitLatch.countDown();
+                assertEquals(i % 100L, rowTuple.<Long>value("key"));
+                assertEquals(i, rowTuple.<Integer>value("valInt"));
+                assertEquals("some string row" + i, rowTuple.<Integer>value("valStr"));
             }
-        });
+        }
+    }
 
-        subscriberAllDataAwaitLatch.await();
+    @Test
+    public void deleteAllOrderTest() throws Exception {
+        var keyRows = populateEvenKeysAndPrepareEntriesToLookup(true);
 
-        assertEquals(0, retrievedItems.size());
+        InternalTable internalTable = ((TableImpl) table).internalTable();
+        var schemaDescriptor = ((TableImpl) table).schemaView().schema();
 
-        populateData(node, keyValueView, implicit);
+        List<BinaryRow> res = internalTable.deleteAll(keyRows, null).get();
+
+        var resIter = res.iterator();
 
-        res = internalTable.scan(0, node.clock().now(), node.node());
+        for (BinaryRowEx key : keyRows) {
+            int i = TableRow.keyTuple(Row.wrapKeyOnlyBinaryRow(schemaDescriptor, key)).<Long>value("key").intValue();
 
-        var subscriberAllDataAwaitLatch2 = new CountDownLatch(1);
+            if (i % 2 == 1) {
+                var rowTuple = TableRow.keyTuple(Row.wrapKeyOnlyBinaryRow(schemaDescriptor, resIter.next()));
 
-        res.subscribe(new Subscriber<>() {
-            @Override
-            public void onSubscribe(Subscription subscription) {
-                subscription.request(10000);
+                assertEquals(i % 100L, rowTuple.<Long>value("key"));
             }
+        }
+    }
 
-            @Override
-            public void onNext(BinaryRow item) {
-                retrievedItems.add(item);
+    @Test
+    public void deleteAllExactOrderTest() throws Exception {
+        var rowsToLookup = populateEvenKeysAndPrepareEntriesToLookup(false);
+
+        InternalTable internalTable = ((TableImpl) table).internalTable();
+        var schemaDescriptor = ((TableImpl) table).schemaView().schema();
+
+        List<BinaryRow> res = internalTable.deleteAllExact(rowsToLookup, null).get();
+
+        var resIter = res.iterator();
+
+        for (BinaryRowEx key : rowsToLookup) {
+            int i = TableRow.tuple(Row.wrapBinaryRow(schemaDescriptor, key)).<Long>value("key").intValue();
+
+            if (i % 2 == 1) {
+                var rowTuple = TableRow.tuple(Row.wrapBinaryRow(schemaDescriptor, resIter.next()));
```

Comment:
It is incorrect to use `var` because the type is not visible from the line of code.